recipe/cross_compile_support.sh

```diff
@@ -15,19 +17,66 @@ if [[ "${HOST_PLATFORM}" != "${BUILD_PLATFORM}" ]]; then
-                    mamba install "${pkg}-cos7-${HOST_PLATFORM:6}" -n sysroot_${HOST_PLATFORM} --yes --quiet || true
+                    mamba install "${pkg}-cos7-${HOST_PLATFORM_ARCH}" -n sysroot_${HOST_PLATFORM} --yes --quiet || true
                 fi
             done
         fi
         export QEMU_LD_PREFIX=$(find ${CONDA_PREFIX}/envs/sysroot_${HOST_PLATFORM} -name sysroot | head -1)
         if [ -f ${CI_SUPPORT}/${CONFIG}.yaml ]; then
             echo "CMAKE_CROSSCOMPILING_EMULATOR: " >> ${CI_SUPPORT}/${CONFIG}.yaml
-            echo "- /usr/bin/qemu-$(echo $HOST_PLATFORM | cut -b 7-)-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+            echo "- /usr/bin/qemu-${HOST_PLATFORM_ARCH}-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
             echo "CROSSCOMPILING_EMULATOR: " >> ${CI_SUPPORT}/${CONFIG}.yaml
-            echo "- /usr/bin/qemu-$(echo $HOST_PLATFORM | cut -b 7-)-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+            echo "- /usr/bin/qemu-${HOST_PLATFORM_ARCH}-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+        fi
+
+
+        if [[ "${CUDA_COMPILER_VERSION}" == "11.2" ]]; then
+            EXTRACT_DIR=$(mktemp -d)
+            pushd ${EXTRACT_DIR}
+                if [[ "${HOST_PLATFORM_ARCH}" == "aarch64" ]]; then
+                    CUDA_HOST_PLATFORM_ARCH=sbsa
+                else
+                    CUDA_HOST_PLATFORM_ARCH=${HOST_PLATFORM_ARCH}
+                fi
+                for f in cuda-cudart-11-2-11.2.72-1 \
+                        cuda-cudart-devel-11-2-11.2.72-1\
+                        cuda-cupti-11-2-11.2.152-1 \
+                        cuda-driver-devel-11-2-11.2.152-1 \
+                        cuda-nvcc-11-2-11.2.152-1 \
+                        cuda-nvml-devel-11-2-11.2.67-1 \
+                        cuda-nvprof-11-2-11.2.67-1 \
+                        cuda-nvrtc-11-2-11.2.152-1 \
+                        cuda-nvrtc-devel-11-2-11.2.152-1 \
+                        cuda-nvtx-11-2-11.2.67-1 \
+                        libcublas-11-2-11.3.1.68-1 \
+                        libcublas-devel-11-2-11.3.1.68-1 \
+                        libcufft-11-2-10.4.1.152-1 \
+                        libcufft-devel-11-2-10.4.1.152-1 \
+                        libcurand-11-2-10.2.3.152-1 \
+                        libcurand-devel-11-2-10.2.3.152-1 \
+                        libcusolver-11-2-11.1.0.152-1 \
+                        libcusolver-devel-11-2-11.1.0.152-1 \
+                        libcusparse-11-2-11.4.1.1152-1 \
+                        libcusparse-devel-11-2-11.4.1.1152-1 \
+                        libnpp-11-2-11.2.1.68-1 \
+                        libnpp-devel-11-2-11.2.1.68-1 \
+                        libnvjpeg-11-2-11.4.0.152-1 \
+                        libnvjpeg-devel-11-2-11.4.0.152-1 ; do
+                    curl -L -O https://developer.download.nvidia.com/compute/cuda/repos/rhel8/${CUDA_HOST_PLATFORM_ARCH}/${f}.${HOST_PLATFORM_ARCH}.rpm
```

Comment:
Is there any potential issues with glibc versioning with using rhel8 cross compilation libraries in a cos7 (or even cos6) sysroot?

Reply:
No. Only constraint is that we have to match the distro we use for native compiling which is centos 8 for cuda>=11.2 for ppc64le and aarch64.